src/catalog/src/durable/impls/persist/metrics.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Prometheus monitoring metrics.
+
+use mz_ore::metric;
+use mz_ore::metrics::{IntCounter, MetricsRegistry};
+use prometheus::{Counter, IntCounterVec};
+
+#[derive(Debug, Clone)]
+pub struct Metrics {
+    pub transactions: IntCounter,
+    pub transaction_commit_errors: IntCounterVec,
+    pub transaction_commit_latency_duration_seconds: Counter,
+    pub transactions_committed: IntCounter,
+    pub snapshot_latency_duration_seconds: Counter,
+    pub snapshots_taken: IntCounter,
+    pub sync_latency_duration_seconds: Counter,
+    pub syncs: IntCounter,
+}
+
+impl Metrics {
+    /// Returns a new [Metrics] instance connected to the given registry.
+    pub fn new(registry: &MetricsRegistry) -> Self {
+        Self {
+            transactions: registry.register(metric!(
+                name: "catalog_transactions",
```

Comment:
Ugh, hate to say this after you've merged, but I think we're pretty good about prefixing all the metrics in db code with `mz_`. Probably worth a followup, or just sneaking it into some other upcoming PR

Reply:
No worries, that's completely my fault for being impatient. I'll post a follow-up.